File: src/components/JSONHandler/src/ALRPCObjectsImpl/TextField.cpp
#include "../include/JSONHandler/ALRPCObjects/TextField.h"
#include "TextFieldMarshaller.h"
#include "CharacterSetMarshaller.h"
#include "TextFieldNameMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	1.2
  date		2011-05-17
  generated at	Wed Nov  7 13:10:41 2012
  source stamp	Thu Oct 25 06:49:27 2012
  author	robok0der
*/



using namespace AppLinkRPC;

TextField::TextField(const TextField& c)
{
  *this=c;
}


bool TextField::checkIntegrity(void)
{
  return TextFieldMarshaller::checkIntegrity(*this);
}


TextField::TextField(void)
{
}



bool TextField::set_characterSet(const CharacterSet& characterSet_)
{
  if(!CharacterSetMarshaller::checkIntegrityConst(characterSet_))   return false;
  characterSet=characterSet_;
  return true;
}

bool TextField::set_name(const TextFieldName& name_)
{
  if(!TextFieldNameMarshaller::checkIntegrityConst(name_))   return false;
  name=name_;
  return true;
}

bool TextField::set_rows(unsigned int rows_)
{
  if(rows_>3)  return false;
  if(rows_<1)  return false;
  rows=rows_;
  return true;
}

bool TextField::set_width(unsigned int width_)
{
  if(width_>40)  return false;
  if(width_<1)  return false;
  width=width_;
  return true;
}




const CharacterSet& TextField::get_characterSet(void) const 
{
  return characterSet;
}


const TextFieldName& TextField::get_name(void) const 
{
  return name;
}


unsigned int TextField::get_rows(void) const
{
  return rows;
}


unsigned int TextField::get_width(void) const
{
  return width;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/TextField.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.CharacterSet;
import com.ford.syncV4.proxy.rpc.enums.TextFieldName;
import com.ford.syncV4.util.DebugTool;

public class TextField extends RPCStruct {

    public TextField() { }
    public TextField(Hashtable hash) {
        super(hash);
    }
    public TextFieldName getName() {
        Object obj = store.get(Names.name);
        if (obj instanceof TextFieldName) {
            return (TextFieldName) obj;
        } else if (obj instanceof String) {
            TextFieldName theCode = null;
            try {
                theCode = TextFieldName.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.name, e);
            }
            return theCode;
        }
        return null;
    }
    public void setName( TextFieldName name ) {
        if (name != null) {
            store.put(Names.name, name );
        }
    }
    public CharacterSet getCharacterSet() {
        Object obj = store.get(Names.characterSet);
        if (obj instanceof CharacterSet) {
            return (CharacterSet) obj;
        } else if (obj instanceof String) {
            CharacterSet theCode = null;
            try {
                theCode = CharacterSet.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.characterSet, e);
            }
            return theCode;
        }
        return null;
    }
    public void setCharacterSet( CharacterSet characterSet ) {
        if (characterSet != null) {
            store.put(Names.characterSet, characterSet );
        }
    }
    public Integer getWidth() {
        return (Integer) store.get( Names.width );
    }
    public void setWidth( Integer width ) {
        if (width != null) {
            store.put(Names.width, width );
        }
    }
    public Integer getRows() {
        return (Integer) store.get( Names.rows );
    }
    public void setRows( Integer rows ) {
        if (rows != null) {
            store.put(Names.rows, rows );
        }
    }
}
